server/api/utils/projects/member.py

```diff
@@ -40,12 +40,14 @@ def ensure_project(
         wait_for_completion: bool = True,
         auth_info: mlrun.common.schemas.AuthInfo = mlrun.common.schemas.AuthInfo(),
     ):
-        project_names = self.list_projects(
+        project = self.get_project(
             db_session,
             format_=mlrun.common.formatters.ProjectFormat.name_only,
             leader_session=auth_info.session,
+            from_leader=False,
+            name=name,
         )
-        if name not in project_names.projects:
+        if not project:
             raise mlrun.errors.MLRunNotFoundError(f"Project {name} does not exist")
```

Comment:
I agree we can keep this but add a comment that `get_project` should raise 404 but we keep this for sanity